src/lib/update-alternatives/UI/main_dialog.rb

```diff
@@ -74,6 +91,27 @@ def map_alternatives_items
       end
     end
 
+    def filtered_alternatives
+      alternatives = @alternatives_list.select { |a| Regexp.new(@search).match(a.name) }
```

Comment:
do we really want to allow full power of regexp for searching? ( just question, not complain :wink: )